### PR TITLE
added dp alogorithm longest common subsequence in dart

### PR DESCRIPTION
**Pull request by RohitSingh107**
# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x ] This pull request is all my own work -- I have not plagiarized.
* [ x] I know that pull requests will not be merged if they fail the automated tests.
* [ x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ x] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/longest_common_subsequence.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import 'dart:math';
+import 'package:test/test.dart';
+
+// Prints the length of the longest common subsequence
+int longest_common_subsequence(String t1, String t2) {
```
Comment (akashgk) [inline dynamic_programming/longest_common_subsequence.dart:5]: improper function naming as per dart convention

File: dynamic_programming/longest_common_subsequence.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  List<List<int>> dp = List<List<int>>.filled(
+      m + 1, List<int>.filled(n + 1, 0, growable: false),
+      growable: false);
+
+  for (int i = 1; i <= m; i++) {
+    for (int j = 1; j <= n; j++) {
+      // if (i == 0 || j == 0) {
+      //   dp[i][j] = 0;
+      //   continue;
+      // }
```
Comment (akashgk) [inline dynamic_programming/longest_common_subsequence.dart:19]: Remove unwanted comments.

File: dynamic_programming/longest_common_subsequence.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import 'dart:math';
+import 'package:test/test.dart';
+
+// Prints the length of the longest common subsequence
+int longest_common_subsequence(String t1, String t2) {
+  int m = t1.length;
+  int n = t2.length;
```
Comment (akashgk) [inline dynamic_programming/longest_common_subsequence.dart:7]: handle edge cases.

File: dynamic_programming/longest_common_subsequence.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import 'dart:math';
```
Comment (akashgk) [inline dynamic_programming/longest_common_subsequence.dart:1]: Can you add links to GFG or Wikipedia Article for the LCS problem

**Review by akashgk — CHANGES_REQUESTED** [review]
Have requested some minor changes. Logic yet to be reviewed

**Comment by RohitSingh107** [comment]
@akashgk I have did all requested changes can you please try again?

File: dynamic_programming/longest_common_subsequence.dart (hunk 3 above)
Comment (RohitSingh107) [reply dynamic_programming/longest_common_subsequence.dart]: I have handled all  edge cases now it's passing all test cases on leetcode.

image

**Comment by RohitSingh107** [comment]
@akashgk why pull request is not progressed, something to change?

**Comment by akashgk** [comment]
> @akashgk why pull request is not progressed, something to change?

@RohitSingh107 apologies for the delay, However please note that these things can take time as we are also involved in other projects and full time work.

**Review by akashgk — APPROVED** [review]
